src/pages/search/index.tsx

```diff
@@ -0,0 +1,194 @@
+  ]);
+
+  useEffect(() => {
+    const handleStart = (url: string) => {
+      if (url !== router.asPath) {
+        document.body.style.cursor = 'wait';
+        setLoading(true);
+      }
+    };
+
+    const handleComplete = (url: string) => {
+      if (url === router.asPath) {
+        document.body.style.cursor = 'auto';
+        setLoading(false);
+      }
+    };
+
+    router.events.on('routeChangeStart', handleStart);
+    router.events.on('routeChangeComplete', handleComplete);
+    router.events.on('routeChangeError', handleComplete);
+
+    return () => {
+      router.events.off('routeChangeStart', handleStart);
+      router.events.off('routeChangeComplete', handleComplete);
+      router.events.off('routeChangeError', handleComplete);
+    };
+  }, [router]);
+
+  useEffect(() => {
+    debouncedServerSearch(startTransition, router, query);
+    setLoading(false);
+    return () => {
+      debouncedServerSearch.cancel();
+    };
+  }, [query]);
+
+  useEffect(() => {
+    posthog.capture('search', {
+      query,
+      count,
+      status: statusFilters.filter((f) => f.checked).map((f) => f.value),
+      skills: skillsFilters.filter((f) => f.checked).map((f) => f.value),
+    });
+  }, []);
+
+  return (
+    <Default
+      meta={
+        <Meta
+          title={`Search - ${query} | Superteam Earn`}
+          description={`Search Results for ${query}`}
+        />
+      }
+    >
+      <Box w="full" minH="100vh" bg="white">
+        <Flex
+          gap={8}
+          w="full"
+          maxW={'7xl'}
+          mx="auto"
+          px={{ base: 3, md: 4 }}
+          py={{ base: 4 }}
+        >
+          <VStack align="start" w={{ base: 'full', md: '70%' }}>
+            <QueryInput loading={loading} query={query} setQuery={setQuery} />
+            <Info loading={loading} count={count} query={query} />
+            <Box display={{ md: 'none' }} w="full">
+              <Filters
+                loading={loading}
+                query={query}
+                statusFilters={statusFilters}
+                skillsFilters={skillsFilters}
+              />
+            </Box>
+            <Divider mx={{ base: 3, md: 4 }} />
+            <Results
+              query={query}
+              results={results}
+              setResults={setResults}
+              count={count}
+            />
+          </VStack>
+          <Box
+            display={{ base: 'none', md: 'block' }}
+            w={{ base: 'full', md: '30%' }}
+          >
+            <Filters
+              query={query}
+              loading={loading}
+              statusFilters={statusFilters}
+              skillsFilters={skillsFilters}
+            />
+          </Box>
+        </Flex>
+      </Box>
+    </Default>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async ({
+  query,
+  req,
+}) => {
+  const protocol = req.headers['x-forwarded-proto'] || 'http';
+  const host = req.headers.host;
+  const fullUrl = `${protocol}://${host}`;
```

Comment:
can use the `getURL` util we have instead

Reply:
ok